lightning/src/ln/peers/handshake/acts.rs

```diff
@@ -1,42 +1,171 @@
-pub const ACT_ONE_LENGTH: usize = 50;
-pub const ACT_TWO_LENGTH: usize = 50;
-pub const ACT_THREE_LENGTH: usize = 66;
-
-/// Wrapper for the first act message
-pub struct ActOne(
-	pub(super) [u8; ACT_ONE_LENGTH]
-);
+//! Helper library for working with NOISE handshake Act data. Contains utilities for passing Act
+//! objects around and building them from received data.
+//! Act objects are thin wrappers about raw arrays for stack-based processing as well as convenient
+//! coercion to slices for flexible use by the higher-level modules.
 
-/// Wrapper for the second act message
-pub struct ActTwo(
-	pub(super) [u8; ACT_TWO_LENGTH]
-);
+use std::{cmp, ops};
 
-/// Wrapper for the third act message
-pub struct ActThree(
-	pub(super) [u8; ACT_THREE_LENGTH]
-);
+pub const ACT_ONE_TWO_LENGTH: usize = 50;
+pub const ACT_THREE_LENGTH: usize = 66;
+pub const EMPTY_ACT_ONE: ActOne = [0; ACT_ONE_TWO_LENGTH];
+pub const EMPTY_ACT_TWO: ActTwo = EMPTY_ACT_ONE;
+pub const EMPTY_ACT_THREE: ActThree = [0; ACT_THREE_LENGTH];
+type ActOne = [u8; ACT_ONE_TWO_LENGTH];
+type ActTwo = ActOne;
+type ActThree = [u8; ACT_THREE_LENGTH];
 
 /// Wrapper for any act message
-pub enum Act {
+pub(super) enum Act {
 	One(ActOne),
 	Two(ActTwo),
-	Three(ActThree),
+	Three(ActThree)
 }
 
 impl Act {
-	/// Convert any act into a byte vector
-	pub fn serialize(&self) -> Vec<u8> {
+	/// Returns the size of the underlying array
+	fn len(&self) -> usize {
+		self.as_ref().len()
+	}
+}
+
+impl From<ActBuilder> for Act {
+	/// Convert a finished ActBuilder into an Act
+	fn from(act_builder: ActBuilder) -> Self {
+		assert!(act_builder.is_finished());
+		act_builder.partial_act
+	}
+}
+
+impl ops::Deref for Act {
+	type Target = [u8];
+
+	/// Allows automatic coercion to slices in function calls
+	/// &Act -> &[u8]
+	fn deref(&self) -> &Self::Target {
 		match self {
 			&Act::One(ref act) => {
-				act.0.to_vec()
+				act
 			}
 			&Act::Two(ref act) => {
-				act.0.to_vec()
+				act
 			}
 			&Act::Three(ref act) => {
-				act.0.to_vec()
+				act
 			}
 		}
 	}
-}
\ No newline at end of file
+}
+
+impl AsRef<[u8]> for Act {
+	/// Allow convenient exposure of the underlying array through as_ref()
+	/// Act.as_ref() -> &[u8]
+	fn as_ref(&self) -> &[u8] {
+		&self
+	}
+}
+
+// Simple fill implementation for both almost-identical structs to deduplicate code
+// $act: Act[One|Two|Three], $input: &[u8]; returns &[u8] of remaining input that was not processed
+macro_rules! fill_impl {
```

Comment:
I think `fill_act_content` might be a bit more descriptive here

Reply:
Resolved in `review: Non-controversial renames, nits, small fixups`